seafile-ccnet compile error

-MT packet-io.o -MD -MP -MF .deps/packet-io.Tpo -c -o packet-io.o `test -f '../common/packet-io.c' || echo './'`../common/packet-io.c
../common/packet-io.c: In function 'canReadWrapper':
../common/packet-io.c:62:5: warning: implicit declaration of function 'EVBUFFER_LENGTH' [-Wimplicit-function-declaration]
     g_return_if_fail (EVBUFFER_LENGTH (e->input) >= CCNET_PACKET_LENGTH_HEADER);
     ^
In file included from /home/main/brcm/staging_dir/target-mipsel_mips32_uClibc-0.9.33.2/usr/include/glib-2.0/glibconfig.h:9:0,
                 from /home/main/brcm/staging_dir/target-mipsel_mips32_uClibc-0.9.33.2/usr/include/glib-2.0/glib/gtypes.h:32,
                 from /home/main/brcm/staging_dir/target-mipsel_mips32_uClibc-0.9.33.2/usr/include/glib-2.0/glib/galloca.h:32,
                 from /home/main/brcm/staging_dir/target-mipsel_mips32_uClibc-0.9.33.2/usr/include/glib-2.0/glib.h:30,
                 from ../../include/ccnet/option.h:7,
                 from ../common/common.h:18,
                 from ../common/packet-io.c:3:
../common/packet-io.c:62:41: error: dereferencing pointer to incomplete type
     g_return_if_fail (EVBUFFER_LENGTH (e->input) >= CCNET_PACKET_LENGTH_HEADER);
                                         ^
/home/main/brcm/staging_dir/target-mipsel_mips32_uClibc-0.9.33.2/usr/include/glib-2.0/glib/gmacros.h:319:8: note: in definition of macro '_G_BOOLEAN_EXPR'
    if (expr)                                    \
        ^
/home/main/brcm/staging_dir/target-mipsel_mips32_uClibc-0.9.33.2/usr/include/glib-2.0/glib/gmessages.h:368:9: note: in expansion of macro 'G_LIKELY'
      if G_LIKELY(expr) { } else            \
         ^
../common/packet-io.c:70:9: warning: implicit declaration of function 'EVBUFFER_DATA' [-Wimplicit-function-declaration]
         packet = (ccnet_packet *) EVBUFFER_DATA (e->input);
         ^
../common/packet-io.c:70:51: error: dereferencing pointer to incomplete type
         packet = (ccnet_packet *) EVBUFFER_DATA (e->input);
                                                   ^
../common/packet-io.c:77:31: error: dereferencing pointer to incomplete type
         if (EVBUFFER_LENGTH (e->input) - CCNET_PACKET_LENGTH_HEADER < len)
                               ^
../common/packet-io.c:93:9: warning: implicit declaration of function 'evbuffer_drain' [-Wimplicit-function-declaration]
         evbuffer_drain (e->input, len + CCNET_PACKET_LENGTH_HEADER);
         ^
../common/packet-io.c:93:26: error: dereferencing pointer to incomplete type
         evbuffer_drain (e->input, len + CCNET_PACKET_LENGTH_HEADER);
                          ^
../common/packet-io.c:95:29: error: dereferencing pointer to incomplete type
         if(EVBUFFER_LENGTH(e->input) >= CCNET_PACKET_LENGTH_HEADER)
                             ^
../common/packet-io.c: In function 'ccnet_packet_io_new':
../common/packet-io.c:133:5: warning: implicit declaration of function 'bufferevent_socket_new' [-Wimplicit-function-declaration]
     io->bufev = bufferevent_socket_new (NULL, io->socket, BEV_OPT_CLOSE_ON_FREE);
     ^
../common/packet-io.c:133:59: error: 'BEV_OPT_CLOSE_ON_FREE' undeclared (first use in this function)
     io->bufev = bufferevent_socket_new (NULL, io->socket, BEV_OPT_CLOSE_ON_FREE);
                                                           ^
../common/packet-io.c:133:59: note: each undeclared identifier is reported only once for each function it appears in
../common/packet-io.c:134:5: warning: implicit declaration of function 'bufferevent_setcb' [-Wimplicit-function-declaration]
     bufferevent_setcb (io->bufev, canReadWrapper,
     ^
../common/packet-io.c:136:5: warning: implicit declaration of function 'bufferevent_enable' [-Wimplicit-function-declaration]
     bufferevent_enable (io->bufev, EV_READ | EV_WRITE);
     ^
../common/packet-io.c: In function 'ccnet_packet_io_free':
../common/packet-io.c:205:9: warning: implicit declaration of function 'bufferevent_free' [-Wimplicit-function-declaration]
         bufferevent_free (io->bufev);
         ^
../common/packet-io.c: In function 'ccnet_packet_io_try_read':
../common/packet-io.c:222:33: error: dereferencing pointer to incomplete type
     if(EVBUFFER_LENGTH(io->bufev->input))
                                 ^
../common/packet-io.c: In function 'ccnet_packet_io_set_timeout_secs':
../common/packet-io.c:250:5: warning: implicit declaration of function 'bufferevent_settimeout' [-Wimplicit-function-declaration]
     bufferevent_settimeout (io->bufev, io->timeout, io->timeout);
     ^
../common/packet-io.c:252:9: warning: implicit declaration of function 'bufferevent_disable' [-Wimplicit-function-declaration]
         bufferevent_disable (io->bufev, EV_READ | EV_WRITE);
         ^
../common/packet-io.c: In function 'ccnet_packet_io_write_packet':
../common/packet-io.c:273:5: warning: implicit declaration of function 'bufferevent_write' [-Wimplicit-function-declaration]
     bufferevent_write (io->bufev, packet, len);
     ^
make[7]: *** [packet-io.o] Error 1
make[7]: Leaving directory `/home/main/brcm/build_dir/target-mipsel_mips32_uClibc-0.9.33.2/seafile-ccnet-4.1.2/net/server'
make[6]: *** [all-recursive] Error 1
make[6]: Leaving directory `/home/main/brcm/build_dir/target-mipsel_mips32_uClibc-0.9.33.2/seafile-ccnet-4.1.2/net'
make[5]: *** [all-recursive] Error 1
make[5]: Leaving directory `/home/main/brcm/build_dir/target-mipsel_mips32_uClibc-0.9.33.2/seafile-ccnet-4.1.2'
make[4]: *** [all] Error 2
make[4]: Leaving directory `/home/main/brcm/build_dir/target-mipsel_mips32_uClibc-0.9.33.2/seafile-ccnet-4.1.2'
make[3]: *** [/home/main/brcm/build_dir/target-mipsel_mips32_uClibc-0.9.33.2/seafile-ccnet-4.1.2/.built] Error 2
make[3]: Leaving directory `/home/main/brcm/feeds/packages/net/seafile-ccnet'
make[2]: *** [package/feeds/packages/seafile-ccnet/compile] Error 2
make[2]: Leaving directory `/home/main/brcm'
make[1]: *** [/home/main/brcm/staging_dir/target-mipsel_mips32_uClibc-0.9.33.2/stamp/.package_compile] Error 2
make[1]: Leaving directory `/home/main/brcm'
make: *** [world] Error 2

--- a/net/seafile-ccnet/patches/010-libevent-include-path.patch
+++ b/net/seafile-ccnet/patches/010-libevent-include-path.patch
@@ -1,0 +1,239 @@
+diff -rupN seafile-ccnet-4.1.2.orig/include/ccnet/ccnet-client.h seafile-ccnet-4.1.2/include/ccnet/ccnet-client.h
+--- seafile-ccnet-4.1.2.orig/include/ccnet/ccnet-client.h	2015-09-03 21:43:37.000000000 +0200
++++ seafile-ccnet-4.1.2/include/ccnet/ccnet-client.h	2015-09-04 10:21:05.578937942 +0200
+@@ -10,11 +10,7 @@
+ #include <glib.h>
+ #include <glib-object.h>
+ 
+-#if defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__)
+ #include <event2/util.h>
+-#else
+-#include <evutil.h>
+-#endif
+ 
+ #include "ccnet-session-base.h"
+ 
+diff -rupN seafile-ccnet-4.1.2.orig/include/ccnet/cevent.h seafile-ccnet-4.1.2/include/ccnet/cevent.h
+--- seafile-ccnet-4.1.2.orig/include/ccnet/cevent.h	2015-09-03 21:43:37.000000000 +0200
++++ seafile-ccnet-4.1.2/include/ccnet/cevent.h	2015-09-04 10:21:10.498933107 +0200
+@@ -6,13 +6,9 @@
+ #ifndef CEVENT_H
+ #define CEVENT_H
+ 
+-#if defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__)
+ #include <event2/event.h>
+ #include <event2/event_compat.h>
+ #include <event2/event_struct.h>
+-#else
+-#include <event.h>
+-#endif
+ 
+ #include <glib.h>
+ 
+diff -rupN seafile-ccnet-4.1.2.orig/lib/job-mgr.c seafile-ccnet-4.1.2/lib/job-mgr.c
+--- seafile-ccnet-4.1.2.orig/lib/job-mgr.c	2015-09-03 21:43:37.000000000 +0200
++++ seafile-ccnet-4.1.2/lib/job-mgr.c	2015-09-04 10:19:49.323012920 +0200
+@@ -1,11 +1,7 @@
+ /* -*- Mode: C; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- */
+ 
+-#if defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__)
+ #include <event2/event.h>
+ #include <event2/event_compat.h>
+-#else
+-#include <event.h>
+-#endif
+ 
+ #include <string.h>
+ #include <stdlib.h>
+diff -rupN seafile-ccnet-4.1.2.orig/lib/libccnet_utils.h seafile-ccnet-4.1.2/lib/libccnet_utils.h
+--- seafile-ccnet-4.1.2.orig/lib/libccnet_utils.h	2015-09-03 21:43:37.000000000 +0200
++++ seafile-ccnet-4.1.2/lib/libccnet_utils.h	2015-09-04 10:20:00.003002414 +0200
+@@ -15,11 +15,7 @@
+ #include <glib-object.h>
+ #include <stdlib.h>
+ 
+-#if defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__)
+ #include <event2/util.h>
+-#else
+-#include <evutil.h>
+-#endif
+ 
+ #ifdef WIN32
+     #include <errno.h>
+diff -rupN seafile-ccnet-4.1.2.orig/lib/mainloop.c seafile-ccnet-4.1.2/lib/mainloop.c
+--- seafile-ccnet-4.1.2.orig/lib/mainloop.c	2015-09-03 21:43:37.000000000 +0200
++++ seafile-ccnet-4.1.2/lib/mainloop.c	2015-09-04 10:20:10.355992232 +0200
+@@ -3,13 +3,9 @@
+ #include "include.h"
+ #include <ccnet.h>
+ 
+-#if defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__)
+ #include <event2/event.h>
+ #include <event2/event_compat.h>
+ #include <event2/event_struct.h>
+-#else
+-#include <event.h>
+-#endif
+ 
+ static int
+ cmdrsp_cb (const char *code, char *content, int clen, void *data)
+diff -rupN seafile-ccnet-4.1.2.orig/lib/net.h seafile-ccnet-4.1.2/lib/net.h
+--- seafile-ccnet-4.1.2.orig/lib/net.h	2015-09-03 21:43:37.000000000 +0200
++++ seafile-ccnet-4.1.2/lib/net.h	2015-09-04 10:20:22.951979845 +0200
+@@ -19,11 +19,7 @@
+     #include <netinet/tcp.h>
+ #endif
+ 
+-#if defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__)
+ #include <event2/util.h>
+-#else
+-#include <evutil.h>
+-#endif
+ 
+ #ifdef WIN32
+     #define ECONNREFUSED WSAECONNREFUSED
+diff -rupN seafile-ccnet-4.1.2.orig/lib/packet-io.h seafile-ccnet-4.1.2/lib/packet-io.h
+--- seafile-ccnet-4.1.2.orig/lib/packet-io.h	2015-09-03 21:43:37.000000000 +0200
++++ seafile-ccnet-4.1.2/lib/packet-io.h	2015-09-04 10:20:31.827971118 +0200
+@@ -5,11 +5,7 @@
+ 
+ #include <packet.h>
+ 
+-#if defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__)
+ #include <event2/util.h>
+-#else
+-#include <evutil.h>
+-#endif
+ 
+ struct buffer;
+ 
+diff -rupN seafile-ccnet-4.1.2.orig/lib/processor.c seafile-ccnet-4.1.2/lib/processor.c
+--- seafile-ccnet-4.1.2.orig/lib/processor.c	2015-09-03 21:43:37.000000000 +0200
++++ seafile-ccnet-4.1.2/lib/processor.c	2015-09-04 10:20:53.395949916 +0200
+@@ -4,11 +4,7 @@
+ 
+ #include <pthread.h>
+ 
+-#if defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__)
+ #include <event2/event.h>
+-#else
+-#include <event.h>
+-#endif
+ 
+ #include "ccnet-client.h"
+ #include "processor.h"
+diff -rupN seafile-ccnet-4.1.2.orig/lib/timer.c seafile-ccnet-4.1.2/lib/timer.c
+--- seafile-ccnet-4.1.2.orig/lib/timer.c	2015-09-03 21:43:37.000000000 +0200
++++ seafile-ccnet-4.1.2/lib/timer.c	2015-09-04 10:20:38.571964488 +0200
+@@ -1,12 +1,8 @@
+ /* -*- Mode: C; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- */
+ 
+-#if defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__)
+ #include <event2/event.h>
+ #include <event2/event_compat.h>
+ #include <event2/event_struct.h>
+-#else
+-#include <event.h>
+-#endif
+ 
+ #include <sys/time.h>
+ 
+diff -rupN seafile-ccnet-4.1.2.orig/lib/utils.h seafile-ccnet-4.1.2/lib/utils.h
+--- seafile-ccnet-4.1.2.orig/lib/utils.h	2015-09-03 21:43:37.000000000 +0200
++++ seafile-ccnet-4.1.2/lib/utils.h	2015-09-04 10:20:45.578957600 +0200
+@@ -11,11 +11,7 @@
+ #include <glib-object.h>
+ #include <stdlib.h>
+ 
+-#if defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__)
+ #include <event2/util.h>
+-#else
+-#include <evutil.h>
+-#endif
+ 
+ #ifdef WIN32
+ #include <errno.h>
+diff -rupN seafile-ccnet-4.1.2.orig/net/common/connect-mgr.h seafile-ccnet-4.1.2/net/common/connect-mgr.h
+--- seafile-ccnet-4.1.2.orig/net/common/connect-mgr.h	2015-09-03 21:43:37.000000000 +0200
++++ seafile-ccnet-4.1.2/net/common/connect-mgr.h	2015-09-04 10:21:20.266923508 +0200
+@@ -3,11 +3,7 @@
+ #ifndef CCNET_CONNECTION_MANAGER
+ #define CCNET_CONNECTION_MANAGER
+ 
+-#if defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__)
+ #include <event2/event.h>
+-#else
+-#include <event.h>
+-#endif
+ 
+ #include "timer.h"
+ 
+diff -rupN seafile-ccnet-4.1.2.orig/net/common/packet-io.c seafile-ccnet-4.1.2/net/common/packet-io.c
+--- seafile-ccnet-4.1.2.orig/net/common/packet-io.c	2015-09-03 21:43:37.000000000 +0200
++++ seafile-ccnet-4.1.2/net/common/packet-io.c	2015-09-04 10:21:28.691915231 +0200
+@@ -13,13 +13,9 @@
+ 
+ #include <unistd.h>
+ 
+-#if defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__)
+ #include <event2/event.h>
+ #include <event2/bufferevent.h>
+ #include <event2/bufferevent_struct.h>
+-#else
+-#include <event.h>
+-#endif
+ 
+ #include <glib.h>
+ #include <errno.h>
+diff -rupN seafile-ccnet-4.1.2.orig/net/common/packet-io.h seafile-ccnet-4.1.2/net/common/packet-io.h
+--- seafile-ccnet-4.1.2.orig/net/common/packet-io.h	2015-09-03 21:43:37.000000000 +0200
++++ seafile-ccnet-4.1.2/net/common/packet-io.h	2015-09-04 10:21:51.130893188 +0200
+@@ -5,13 +5,9 @@
+ 
+ #include "packet.h"
+ 
+-#if defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__)
+ #include <event2/util.h>
+ #include <event2/buffer.h>
+ #include <event2/buffer_compat.h>
+-#else
+-#include <evutil.h>
+-#endif
+ 
+ /* struct evbuffer; */
+ /* for libevent2 */
+diff -rupN seafile-ccnet-4.1.2.orig/net/common/peer.c seafile-ccnet-4.1.2/net/common/peer.c
+--- seafile-ccnet-4.1.2.orig/net/common/peer.c	2015-09-03 21:43:37.000000000 +0200
++++ seafile-ccnet-4.1.2/net/common/peer.c	2015-09-04 10:22:00.506883980 +0200
+@@ -2,14 +2,10 @@
+ 
+ #include "common.h"
+ 
+-#if defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__)
+ #include <event2/event.h>
+ #include <event2/bufferevent.h>
+ #include <event2/bufferevent_compat.h>
+ #include <event2/bufferevent_struct.h>
+-#else
+-#include <event.h>
+-#endif
+ 
+ #include <stdio.h>
+ #include <stdlib.h>
+diff -rupN seafile-ccnet-4.1.2.orig/net/common/session.h seafile-ccnet-4.1.2/net/common/session.h
+--- seafile-ccnet-4.1.2.orig/net/common/session.h	2015-09-03 21:43:37.000000000 +0200
++++ seafile-ccnet-4.1.2/net/common/session.h	2015-09-04 10:22:13.114871598 +0200
+@@ -3,13 +3,9 @@
+ #ifndef CCNET_SESSION_H
+ #define CCNET_SESSION_H
+ 
+-#if defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__)
+ #include <event2/event.h>
+ #include <event2/event_compat.h>
+ #include <event2/event_struct.h>
+-#else
+-#include <event.h>
+-#endif
+ 
+ #include <glib.h>
+ #include <glib/gstdio.h>

